sktime/data/pbf.py

```diff
@@ -0,0 +1,312 @@
+    try to reach by a smoothed particle hydrodynamics style simulation [1]_ [2]_.
+
+    In some more detail:
+
+    Each particle has positions :math:`p_1,\ldots,p_n \in\mathbb{R}^d` and
+    velocity :math:`v_1,\ldots,v_n\in\mathbb{R^d}`. For the local density a standard SPH estimator is used
+
+    .. math ::
+        \rho_i = \sum_i m_i W(p_i - p_j, h),
+
+    where :math:`m_i` refers to the particle's mass (here :math:`m_i \equiv 1` for simplicity), :math:`W` is a
+    convolution kernel with interaction distance :math:`h`, i.e., particles with distance larger than :math:`h`
+    get assigned a weight of zero in the convolution operation.
+
+    The whole system has an underlying family of density constraints
+
+    .. math ::
+        C_i(p_1,\ldots,p_n) = \frac{\rho_i}{\rho_0} - 1.
+
+    The simulation algorithm tries to fulfill this constraint by updating each particle's position, i.e., it tries to
+    find :math:`\Delta p_k` so that
+
+    .. math ::
+        C_k(p_1,\ldots,p_k + \Delta p_k,\ldots,p_n) = 0.
+
+    This is done by performing Newton's optimization method along the constraint's gradient. Because of potentially
+    vanishing gradients, the optimization is regularized / dampened with a parameter :math:`\varepsilon`. In this
+    implementation, larger values of :math:`\varepsilon` lead to more regularization.
+
+    When a particle has only very few neighbors, it might not be possible to reach the rest density. This can lead
+    to particle clumping (i.e., areas of negative pressures). One remedy is an artificial internal pressure term,
+    here referred to as tensile instability. It modifies the weighting inside the convolution kernel itself.
+    The default parameters `tensile_instability_distance` and `tensile_instability_k` are the suggested ones in the
+    PBF publication.
+
+    References
+    ----------
+    .. [1] Gingold, Robert A., and Joseph J. Monaghan. "Smoothed particle hydrodynamics: theory and application
+           to non-spherical stars." Monthly notices of the royal astronomical society 181.3 (1977): 375-389.
+    .. [2] Lucy, Leon B. "A numerical approach to the testing of the fission hypothesis." The astronomical
+          journal 82 (1977): 1013-1024.
+    """
+
+    def __init__(self, domain_size: np.ndarray, initial_positions: np.ndarray,
+                 interaction_distance: float, n_jobs=None, n_solver_iterations: int = 5,
+                 gravity: float = 10., epsilon: float = 10., timestep: float = 0.016, rest_density: float = 1.,
+                 tensile_instability_distance: float = .2, tensile_instability_k: float = 0.1):
+        r"""Creates a new simulator instance.
+
+        Parameters
+        ----------
+        domain_size : (2,) ndarray
+            A 1-dimensional ndarray with two elements describing the extension of the simulation box.
+        initial_positions : (N, 2) ndarray
+            A 2-dimensional ndarray describing the :math:`N` particles' initial positions.
```

Comment:
this is exactly how the user specifies number of particles, right? by the shape of this array?

Reply:
yes, added clarifying docstring

Reply:
thank you!